### ISSUE TITLE
NPE exception when using equalToDateTime matcher with truncate and actualFormat with just the day part of the date 

### ISSUE THREAD

**Issue by toniisidoro** [wiremock/wiremock#1711]
Null Pointer Exception when using equalToDateTime matcher with truncate and actualFormat with just the day part of the date, example:

Given an API request payload using actual date as separated fields (year, month and day):
```Json
{
  "startDate": {
    "year": 2021,
    "month": 11,
    "day": 25
  }
}
```
When I create the stub using the request matching below:
```Json
 {
   "request" : {
    "url" : "/path",
    "method" : "POST",
    "bodyPatterns" : [
    {
      "matchesJsonPath" : {
        "expression" : "$.startDate.day",
        "equalToDateTime": "now +1 days",   
        "actualFormat" : "dd",
        "truncateActual": "first hour of day"
      }
    }
 ]
  }
```
Then it should match a request correctly

- [x]  **WireMock version:** 2.31.0
- [x] **How:** 
   Run test `com.github.tomakehurst.wiremock.matching.MatchesJsonPathPatternTest#matchesCorrectlyWhenUsingActualFormatWithJustTheDayPartOfTheDateAndTruncatingTheDateToo`
- [x] **Failing test case that demonstrates the problem:** [PR](https://github.com/wiremock/wiremock/pull/1710/files)

**Comment by tomakehurst** [comment]
The issue with this is that `actualFormat` refers to the format used to parse the incoming value, so in the case the `days` field from the request JSON. Since a day on its own doesn't contain enough information to be resolved to a full zoned date, it can't be parsed. The resulting NPE is pretty bad, so that definitely needs to be improved.

I think what this needs is the ability to construct a date from multiple incoming values, as its seems really what you'd like to do is ensure that the entire date represented by day/month/year is tomorrow. Is that assumption correct?

**Comment by toniisidoro** [comment]
yes, it should just match the request by verifying if the startDate field is in fact a compound object that represents the tomorrow date.
